provider/pkg/provider/kind/kind.go

```diff
@@ -0,0 +1,243 @@
+// File: provider/pkg/provider/kind/kind.go
+
+// Package provider encapsulates logic for managing Kind clusters using Pulumi.
+package provider
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/pulumi/pulumi/sdk/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	local "github.com/pulumi/pulumi/sdk/v3/go/pulumi/cmd"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/provider/cmd/pulumi-resource-command/schema"
+
+	// Import version package to obtain the current provider version.
+	pv "github.com/usrbinkat/pulumi-kind-native/provider/pkg/version"
+)
+
+// kindProviderVersion holds the version information of this custom provider.
+var kindProviderVersion = pv.GetVersion()
+
+// CreateKindClusterArgs constructs and returns a KindClusterArgs struct
+// based on the given PropertyMap inputs.
+// Provides default values for unspecified properties.
+func CreateKindClusterArgs(inputs resource.PropertyMap) (KindClusterArgs, error) {
+	// Initialize arguments with default values.
+	clusterName := "pulumi"
```

Comment:
Defaults should be applied in `Check`. I would worry about this after you get the basic provider working, but it will provide a nicer user experience for users when they consume the provider.